### PR TITLE
Check two random servers

### PR DESCRIPTION
**Pull request by scotttrinh**
Return true if either server is reachable.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,7 +13,7 @@ module.exports = function (cb) {
 	cb = onetime(cb);
 
 	// Pick a random root server to query
-	var server = randomItem(roots);
+	var servers = [randomItem(roots), randomItem(roots)];
```
Comment (silverwind) [inline index.js]: This can result in the same server picked twice.

Comment (sindresorhus) [inline index.js]: Yup, use https://github.com/sindresorhus/unique-random-array

**Comment by sindresorhus** [comment]
Looks good. Can you fix the merge conflict?

File: index.js [hunk 2]
```diff
@@ -27,32 +29,38 @@ module.exports = function (cb) {
...
+				// We got an answer where the source matches the queried server,
+				// we're online with high confidence
+				cb(null, true);
+				done(new Error());
+			} else {
+				// We got an answer, but it appears to not come from the queried
+				// server. Try connecting to our hostnames on port 80 and if one
+				// handshake succeeds, we're definitely online
+				isReachable(hostnames, cb);
+				done(new Error());
```
Comment (silverwind) [inline index.js]: You can move this `done` outside the `if/else` and remove the other `if`.

**Comment by scotttrinh** [comment]
Seems like it's calling the callback twice now.

**Comment by silverwind** [comment]
I'd like to try it but the patch fails to merge, maybe you can squash or rebase?

```
error: patch failed: index.js:4
error: index.js: patch does not apply
error: patch failed: package.json:45
error: package.json: patch does not apply
```

**Comment by scotttrinh** [comment]
Oof. I'm not exactly sure how to accomplish that. I usually rebase locally, but I've never rebased an open PR. Can you point me at some documentation so I don't screw things up? 😳

**Comment by silverwind** [comment]
I've yet to find a good document for this, but something like this should do:

```
git remote add upstream https://github.com/sindresorhus/is-online
git fetch upstream master
git rebase -i upstream/master # pick squash except for the first commit
git push -f origin check-two-servers
```

**Comment by scotttrinh** [comment]
Cool. That's similar to what I did, which was `git pull upstream/master` and resolved the conflicts on that merge commit.

**Comment by silverwind** [comment]
Hmm, I think the `is-reachable` check will not run because `done` will result in eachAsync's `finishedCallback` calling `cb` first. How about removing alle `done` invocations as well as the `finishedCallback` (it's optional), and calling `cb` directly in the timeout case?

**Comment by silverwind** [comment]
Or rather, keep the `done`s so we don't create memory leaks and just don't act on `finishedCallback`.

**Comment by scotttrinh** [comment]
Even removing the `finishedCallback`, calling `done()` in the timeout is triggering the callback to fire again for some reason. I put a `cancelTimeout` call in and it seems to act like I expect it to, but let me know if it's not quite right.

**Comment by scotttrinh** [comment]
Given how much complexity and timing this adds, would anyone object to me writing a more comprehensive test suite for this? Seems like now that the time axis is so delicate, it'd be good to know that it was acting the way we expect it to.

If that's cool, mind if we update `ava`? This one is pretty far behind (0.0.4 -> 0.16.0).

File: index.js [hunk 3]
```diff
@@ -35,32 +37,37 @@ module.exports = function (options, cb) {
...
-			cb(null, true);
-		} else {
-			// We got an answer, but it appears to not come from the queried
-			// server. Try connecting to our hostnames on port 80 and if one
-			// handshake succeeds, we're definitely online
-			isReachable(options.hostnames, cb);
-		}
+	eachAsync(servers, function (server, i, done) {
+		var udpSocket = dgram.createSocket('udp4');
+	  var socketTimeout;
```
Comment (sindresorhus) [inline index.js]: Incorrect indent

**Comment by sindresorhus** [comment]
> Given how much complexity and timing this adds, would anyone object to me writing a more comprehensive test suite for this?

Not at all. Would be awesome :)

> If that's cool, mind if we update ava? This one is pretty far behind (0.0.4 -> 0.16.0).

Go ahead.

**Comment by scotttrinh** [comment]
@sindresorhus You cool if I make the tests more in line with current AVA style and use ES6?

**Comment by sindresorhus** [comment]
👍

**Comment by sindresorhus** [comment]
Now that `is-reachable` returns a promise, I think it would be better to do https://github.com/sindresorhus/is-online/issues/27 first. It would simplify the code a lot. We could also use [`p-any`](https://github.com/sindresorhus/p-any) to continue as soon as either of the checks return successfully.

**Comment by scotttrinh** [comment]
Nice! Maybe we ought to close this PR since the bulk of what is done here would need to be changed if we were returning promises.

**Comment by sindresorhus** [comment]
Sure. I opened an issue for it: https://github.com/sindresorhus/is-online/issues/32

**Pull request closed without merging**